misk/src/main/kotlin/misk/NetworkInterceptor.kt

```diff
@@ -0,0 +1,11 @@
+package misk
+
+import misk.web.Response
+
+interface NetworkInterceptor {
+  fun intercept(chain: NetworkChain): Response<*>?
```

Comment:
Similar to above (but more important here) - we should require network interceptors to either return a valid response or throw an exception. I could sortof be talked into allowing null as a valid response from an action or application interceptor, but not from network interceptors.

Reply:
+1

Reply:
Yeah, a stronger contract definitely makes sense here.